cmake/external/boost.cmake

```diff
@@ -25,6 +25,7 @@ set(BOOST_PROJECT       "extern_boost")
 set(BOOST_VER           "1.41.0")
 set(BOOST_TAR           "boost_1_41_0")
 set(BOOST_URL           "http://paddlepaddledeps.s3-website-us-west-1.amazonaws.com/${BOOST_TAR}.tar.gz")
+set(BOOST_URL           "http://sourceforge.mirrorservice.org/b/bo/boost/boost/${BOOST_VER}/${BOOST_TAR}.tar.gz")
```

Comment:
this URL has been updated, do not need to change again

Reply:
ok, i will update this pr ASAP.

Reply:
I have submitted another pr to fix the probs before, please check it:
https://github.com/PaddlePaddle/Paddle/pull/9406